Push notification post (#728)
RocketChat/rocketchat.github.io: _posts/2019/09/2019-09-20-push-notification.md

---

> title: "Push notification"

Author: not sure this is the best title

Reviewer: Maybe "News on Push Notification behavior"?

Author: "Push Notification behavior changes" ?

Reviewer 2: It's not related to Push Notifications only right? It could be `Changes on the behavior of notifications`

Reviewer: "Changes on notifications behavior" then?

categories:

> - News

Author: other categories?

Reviewer 2: Maybe Product

date: 2019-09-20 12:00:00
author: Diego Sampaio

> cover: /images/posts/2019/08/postmortem-android-3.5.1.png

Author: still missing a cover image

Reviewer 2: Suggestion https://www.referralcandy.com/wp-content/uploads/2019/07/push-notification-marketing-referralcandy-guestpost-740x389.png

Author: I have selected a few 'abstract' images that I think aligns with the other images we have used: https://github.com/RocketChat/rocketchat.github.io/tree/push-notification-post/images/posts/2019/09/push

Author: the chosen one https://github.com/RocketChat/rocketchat.github.io/blob/push-notification-post/images/posts/2019/09/push/bell.jpg

featured: true
---

## Current state

> Rocket.Chat has always tried to be smart when sending mobile notifications. We don't want to disturb you with useless notifications, but we realize this is confusing to our users. Users expect notifications and think there are bugs when none are actually being sent.

Reviewer: ```suggestion
Rocket.Chat has always tried to be smart when sending mobile notifications. We don't want to disturb you with useless notifications, but we understand this is confusing to our users. Users expect notifications and think there are bugs when none are actually being sent.
```

Maybe "understand" is better? 🤔

We have created a diagram to show the conditions that cause Rocket.Chat to send a notification:

> <a href="https://whimsical.com/PRwN4MWNsxSwqHjHXaPUuC"><img src="/images/posts/2019/09/push-notification-diagram.png" style="width: 50%"></a>

Author: I'd like this image to be centralized

Reviewer 3: Since it is a blog post, perhaps starts with a statement of impact.  How will it impact /  affect the reader?

Something along the line of :

"Starting with release 2.1.0,  Rocket.Chat users  may be getting notification more frequently than they are used to.   ....."

Author: I like of starting with a message of more impact..

## Coming changes

Starting at Rocket.Chat version **2.1.0**, the default value for the user's default **Desktop Notification Default Alert** and **Mobile Notifications Default Alert** will be **All Messages**. When updating to version 2.1.0 or later the servers that were still using the previous default value **Mentions**, **will have** the value updated to **All Messages**.

Large rooms (with more than 100 members) are still prevented from sending notifications on all messages. There is a setting that can be changed to increase this number called **Maximum Channel Size for ALL Message** (under Admin > Message) but **please use this setting with caution** as it may cause performance issues.

## Future actions

To match the expectations of our users, we know there is room for improvement. So we're already planning to refactor the push notification logic as a whole with the following key things in mind:

* Allow scaling to larger rooms - no need to worry about the "Maximum Channel Size for ALL Message" setting anymore;
* Remove a notification from mobile devices if that message was read on another device;

> * Do not rely on connection status to send a notification, but instead create a queue of notifications and send them if the message is not read in a certain amount of time - no "magic" anymore, you'll always get a notification on your phone if don't read the message.

Reviewer 3: A summary of the impact,  how it will affect/be visible to the end-user, will be helpful for understanding.

Does this mean that if I read my message on my tablet, I will not get notification on my desktop app and my mobile phones (both Android and iOS)?

Author: yes.. that's what this means:
> send them (the notification) if the message is not read in a certain amount of time

do you think it is enough by saying "not read in **any device**" is enough?

Author: yes.. that's what this means:
> send them (the notification) if the message is not read in a certain amount of time

do you think it is enough by saying "not read in **any device**" is enough?

Author: changed to:
> * Do not rely on connection status to send a notification, but instead create a queue of notifications and send them if the messages were not seen in any device for a certain amount of time - no "magic" anymore, you'll always get a notification on your phone if don't read a message.

wdyt?

---

> title: "Changes on notification behavior"

Reviewer 4: ```suggestion
title: "Changes to notification behavior"
```

categories:
- News
- Product
date: 2019-09-20 12:00:00
author: Diego Sampaio
cover: /images/posts/2019/09/push/bell.jpg
featured: true
---

## Current state

Rocket.Chat has always tried to be smart when sending mobile notifications. We don't want to disturb you with useless notifications, but we understand this is confusing to our users. Users expect notifications and think there are bugs when none are actually being sent.

> We have created a diagram to show the conditions that cause Rocket.Chat to send a notification:

Reviewer 4: ```suggestion
We have created a diagram to show the current conditions that cause Rocket.Chat to send a notification:
```

<a href="https://whimsical.com/PRwN4MWNsxSwqHjHXaPUuC"><img src="/images/posts/2019/09/push/diagram.png" style="width: 50%"></a>

The conditions are hard to understand by new Rocket.Chat users, so we decided to change to a more predictable and consistent approach to benefit our users.

## Coming changes

> Starting at Rocket.Chat version **2.1.0**, the default value for the user's default **Desktop Notification Default Alert** and **Mobile Notifications Default Alert** will be **All Messages**. When updating to version 2.1.0 or later the servers that were still using the previous default value **Mentions**, **will have** the value updated to **All Messages**.

Reviewer 4: ```suggestion
Starting at Rocket.Chat version **2.1.0**, the value for **Desktop Notification Default Alert** and **Mobile Notifications Default Alert** will be **All Messages**. When updating to version 2.1.0 or later the servers that were still using the previous default value **Mentions**, **will have** the value updated to **All Messages**.
```

Reviewer 4: Default is in the setting name so I think ok to remove the other two.  Otherwise when reading feels like saying default a lot 😁

Author: indeed

Large rooms (with more than 100 members) are still prevented from sending notifications on all messages. There is a setting that can be changed to increase this number called **Maximum Channel Size for ALL Message** (under Admin > Message) but **please use this setting with caution** as it may cause performance issues.

## Future actions

> To match the expectations of our users, we know there is room for improvement. So we're already planning to refactor the push notification logic as a whole with the following key things in mind:

Reviewer 4: ```suggestion
To match the expectations of our users, we know there is room for improvement. So we are planning to refactor the push notification logic as a whole with the following key things in mind:
```

* Allow scaling to larger rooms - no need to worry about the "Maximum Channel Size for ALL Message" setting anymore;
* Remove a notification from mobile devices if that message was read on another device;

> * Do not rely on connection status to send a notification, but instead create a queue of notifications and send them if the messages were not seen in any device for a certain amount of time - no "magic" anymore, you'll always get a notification on your phone if don't read a message.

Reviewer 4: ```suggestion
* Do not rely on connection status to send a notification, but instead create a queue of notifications and send them if the messages were not seen on any device for a certain amount of time - no "magic" anymore, you'll always get a notification on your phone if don't read a message.
```
